### DISCUSSION TITLE
Welcome to c-ray Discussions!

### DISCUSSION THREAD

**Discussion by vkoskiv** [VKoskiv/c-ray#89] [General]
Feel free to propose ideas, or ask questions here.

**Comment by DorianCorr** [comment]
Love your project. I started as a C/Assembler programmer during the late 70s and always loved C as a main language but made my living with C#... anyway..

I wanted to point out that the "g" tag in an object file is for material groups. I noticed your comment in the source and thought I'd chime in in case you haven't come across this information.
The materials are attached to groups and the groups contain the faces. The vertices, normals and texture coords can be moved to the group once the "polys" "faces" are defined ( with the "f" tag). 

Anyway, I enjoy your C work and especially your SDL2 work as well.  I can be reached at my email if you have any comments or questions. 

Dorian

**Reply by vkoskiv** [reply]
I appreciate the kind words, Dorian! It's a fun coincidence that you were looking at the wavefront loader, I've been working on some pretty significant changes in the `api-wip` branch that include improvements to the wavefront loader, too. The whole thing definitely needs more improvement. It now correctly handles `o` tags, so it generates several meshes instead of shoving all the geometry into a single mesh.
Groups weren't obvious to me yet, so I appreciate the clarification. I'll try to get that branch to a stable state soon, so I can merge it to master. When I get the new API into a good enough state, It'll be possible to integrate c-ray with other projects, like Blender, and move asset loading into separate pluggable modules.

**Comment by DorianCorr** [comment]
Thank you for the response. It sounds like you’re on the case so I will keep an eye out for those changes. Very exciting stuff.

Dorian

Sent from Mail<https://go.microsoft.com/fwlink/?LinkId=550986> for Windows
